### ISSUE TITLE
Dart analyzer cache grows without bounds

### ISSUE THREAD

**Issue by ds84182** [dart-lang/sdk#29508]
I just cleared my _1.3gb_ Dart Analyzer cache yesterday.

_1.3gb_.

I was wondering where all my space went.

Anyways, there should probably be a configurable cap on the cache size. _1.3gb_ is kinda insane.

**Comment by zoechi** [comment]
I also saw that there is a huge amount of files created.

**Comment by ds84182** [comment]
Yeah, my file manager pretty much refused to open the directory, I think I
had about 13,000 files?

**Comment by bwilkerson** [comment]
@scheglov Is this expected (as opposed to "right")?

**Comment by scheglov** [comment]
Yes, this is expected.
`1.3GB` is on a high side, we start eviction isolate after writing `128M` of new data.
OTOH, this isolate might take some time to process files and start deleting them.

FWIW my cache has `1,069,761K bytes in 133689 files`.
It takes a couple of seconds for Midnight Commander to open it.

**Comment by devoncarew** [comment]
As another data-point, my folder is 1.33 GB.

**Comment by devoncarew** [comment]
And 121,425 files!
